src/authtoken/test.py

```diff
@@ -0,0 +1,226 @@
+import datetime
+import jwt
+
+from django.conf.urls import url
+from django.http import HttpResponse
+from django.test import override_settings, TestCase
+from django.utils import timezone
+from rest_framework import status
+from rest_framework.test import APIClient, APIRequestFactory
+from rest_framework.views import APIView
+
+from .authentication import FooCardAuthentication
+from .models import Token
+from .views import obtain_foocard_token
+from foobar.models import Account
+from foobar.tests.factories import AccountFactory, CardFactory
+
+factory = APIRequestFactory()
+
+
+class MockView(APIView):
+    def get(self, request):
+        return HttpResponse({'some': 1, 'thing': 2, 'dandy': 3})
+
+    def post(self, request):
+        return HttpResponse({'some': 1, 'thing': 2, 'dandy': 3})
+
+    def put(self, request):
+        return HttpResponse({'some': 1, 'thing': 2, 'dandy': 3})
+
+
+urlpatterns = [
+    url(r'^cardtoken/$',
+        MockView.as_view(authentication_classes=[FooCardAuthentication])),
+    url(r'^auth-token/$', obtain_foocard_token),
+]
+
+
+@override_settings(ROOT_URLCONF='authtoken.test', SECRET_WEBTOKEN='secret')
+class AuthenticateFooCardTokenTests(TestCase):
+    enc = 'HS256'
+    path = '/cardtoken/'
+
+    def setUp(self):
+        self.csrf_client = APIClient(enforce_csrf_checks=True)
+
+    def test_correct_auth_of_token(self):
+        account = AccountFactory()
+        token = jwt.encode(
+            {'account_id': str(account.pk)},
+            'secret',
+            algorithm=self.enc
+        )
+        auth = 'CardToken {0}'.format(token.decode())
+        response = self.csrf_client.post(
+            self.path,
+            {'example': 'example'},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_auth_with_invalid_token(self):
+        token = jwt.encode(
+            {'some': 'payload'},
+            'invalidsecret',
+            algorithm=self.enc
+        )
+        auth = 'CardToken {0}'.format(token.decode())
+        response = self.csrf_client.post(
+            self.path,
+            {'example': 'example'},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_auth_with_expired_token(self):
```

Comment:
It seems that there a couple of tests that seem to test that the `pyjwt` module works correctly. `pyjwt` seems to be well tested and I think we could simply mock it away and make sure the `decode` function is called with appropriate arguments.